### PR TITLE
Csharp

### PR DESCRIPTION
**Pull request by kirill-andreyev**

### CODE REVIEW

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            string date;
+            date = DateTime.Now.ToString("dd.MM.yyy");
+            StringToDate(date);
+        }
+        public Date(string date)
+        {
+            StringToDate(date);
+        }
+
+        private void StringToDate(string date)
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:19]: Such methods are usually called Parse

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        {
+            day = (date[0] - 48) * 10 + (date[1]-48);
+            month = (date[3] - 48) * 10 + (date[4] - 48);
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:22]: WHAAAT THE FUCK...

Comment (dandreyew) [reply C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:22]: Holy balls...

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        }
+
+        private int MonthCount()
+        {
+			switch (month)
+			{
+				case 1:
+					return 31;
+				case 2:
+					if (year % 4 == 0)
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:38]: That's not how it works actually, use google.com

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        }
+
+        public void DisplayDate()
+        {
+            Console.WriteLine($"Current date: {day}.{ month}.{ year}\n");
+        }
+
+        private int MonthCount()
+        {
+			switch (month)
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:33]: 1. We have a lamda construction for switch in C# which

Comment (dandreyew) [reply C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:33]: 2. Dictionnary is better

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+					return 31;
+				default:
+					return 0;
+			}
+		}
+
+		internal void AddDays()
+        {
+			int addDay, maxdays;
+			Console.WriteLine("How many days you want to add? ");
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:70]: Why the fuck Date class handles console AGAIN???

**Review by dandreyew — CHANGES_REQUESTED** [review]

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs (hunk 3 above)
Comment (lALTAIRl) [reply C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:38]: You need to choose one of 3 cases, not 1 of 12.

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs (hunk 1 above)
Comment (lALTAIRl) [reply C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:19]: Or Convert, anyway method name should contain a verb that discribes an action, that will be done there.

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs (hunk 4 above)
Comment (lALTAIRl) [reply C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:33]: You need to choose from 3 options, not from 12

File: C_SHARP_8115/8115_fixed/8115_fixed/Program.cs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+                    case 1:
+                        date.AddDays();
+                        date.DisplayDate();
+                        break;
+                    case 2:
+                        Console.WriteLine("Exit...");
+                        Environment.Exit(0);
+                        break;
+                    default:
+                        Console.WriteLine("Wrong choise!");
```
Comment (lALTAIRl) [inline C_SHARP_8115/8115_fixed/8115_fixed/Program.cs:28]: "Choice" the correct spelling

File: C_SHARP_8115/8115_fixed/8115_fixed/Program.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+                }
+            }
+        }
+
+        private static Date NewDate()
+        {
+
+            Console.WriteLine("Date type 1. Y M D 2. DD.MM.YYYY Other. Default");
+            int choise = Convert.ToInt32(Console.ReadLine());
+            switch (choise)
```
Comment (lALTAIRl) [inline C_SHARP_8115/8115_fixed/8115_fixed/Program.cs:39]: "Choice" again

File: C_SHARP_8115/8115_fixed/8115_fixed/Program.cs [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+            {
+                case 1:
+                    {
+                        Console.WriteLine("Enter Day, Month, Year ");
+                        int day = Convert.ToInt32(Console.ReadLine());
+                        int month = Convert.ToInt32(Console.ReadLine());
+                        int year = Convert.ToInt32(Console.ReadLine());
+                        Date date = new Date(year, month, day);
```
Comment (lALTAIRl) [inline C_SHARP_8115/8115_fixed/8115_fixed/Program.cs:47]: use var for declaring variables

**Review by lALTAIRl — CHANGES_REQUESTED** [review]

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs (hunk 1 above)
Comment (dandreyew) [reply C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:19]: > Or Convert, anyway method name should contain a verb that discribes an action, that will be done there.

Convert are usually static members

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+namespace _8115_fixed
+{
+    class Date
+    {
+        public Date(int y, int m, int d) { year = y; month = m; day = d; }
+        public Date()
+        {
+            string date;
+            date = DateTime.Now.ToString("dd.MM.yyy");
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:11]: var date = ...

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs (hunk 3 above)
Comment (kirill-andreyev) [reply C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:38]: > That's not how it works actually, use google.com

I didn't quite understand what the problem is here

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            var date = DateTime.Now.ToString("dd.MM.yyy");
+            ParseStringToDate(date);
+        }
+        public Date(string date)
+        {
+            ParseStringToDate(date);
+        }
+
+        private void ParseStringToDate(string date)
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:20]: just parse. We already know the parameter is string, and we already know the class is Date

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            string[] subDate = date.Split(' ', '.');
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:22]: use var for local variables

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+            day = int.Parse(subDate[0]);
+            month = int.Parse(subDate[1]);
+            year = int.Parse(subDate[2]);
+        }
+
+        public void DisplayDate()
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:28]: Third time. This class should handle ONLY date. No console should be here.

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            Console.WriteLine($"Current date: {day}.{ month}.{ year}\n");
+        }
+		internal void AddDays(int addDay)
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:32]: What's wrong with the formatting?

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+        {
+
+            Dictionary<int, int> monthCount = new(12)
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:35]: This is a very strange name for the variable

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+            if (year % 2 == 0)
+            {
+                if(year % 100 == 0)
+                {
+                    if (year % 400 == 0)
+                        return true;
+                    else
+                        return false;
+                }
+                else
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:86]: You don't need else here

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+                {
+                    return true;
+                }
+            }
+            else
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:91]: No need in else here

File: C_SHARP_8115/8115_fixed/8115_fixed/Date.cs [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+                return false;
+        }
+
+        public int year;
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Date.cs:95]: Place fields and properties before methods

File: C_SHARP_8115/8115_fixed/8115_fixed/Program.cs [hunk 18]
```diff
@@ -0,0 +1,69 @@
+﻿using System;
+
+namespace _8115_fixed
+{
+    class Program
+    {
+        static void Main()
+        {
+            Date date;
```
Comment (dandreyew) [inline C_SHARP_8115/8115_fixed/8115_fixed/Program.cs:9]: var date = new Date();

**Pull request closed without merging**